drivers/include/periph/gpio.h

```diff
@@ -72,39 +81,71 @@
 
 #include <limits.h>
 
-#include "periph_cpu.h"
-#include "periph_conf.h"
+#include "gpio_arch.h"  /* include architecture specific GPIO definitions */
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#ifndef HAVE_GPIO_T
+#ifndef HAVE_GPIO_PIN_T
 /**
- * @brief   GPIO type identifier
+ * @brief   GPIO pin number type
  */
-typedef unsigned int gpio_t;
+typedef unsigned int gpio_pin_t;
```

Comment:
How about using `uint_fast8_t` instead of `unsigned int`? The value range of an `uint8_t` should be sufficient and an `uint_fast8_t` can be accessed with a single read/write operation on all platforms. On platforms that can access a single byte with one read/write operation this will also only consume 1 byte RAM, instead of 2-4 byte.

(E.g. `uint_fast8_t` will be the same as an `uint8_t` on AVR, which can be accessed with one read/write (or 2 CPU cylces); an `unsigned int` would be the same as an `uint16_t` on that platform, which needs 2 read/write operations (or 4 CPU cycles).)

Reply:
I will try.